src/libs/actions/App.js

```diff
@@ -252,6 +252,7 @@ function openProfile() {
 }
 
 // When the app reconnects from being offline, fetch all initialization data
+NetworkConnection.listenForReconnect();
```

Comment:
Hmm I'm not sure that we should be calling `listenForReconnect` here, nor `onReconnect` for that matter (my bad 😅 ). The reason being that it also registers our reconnect callbacks even when the user is not logged in! You can see calls to `ReconnectApp` being triggered when you are logged out and come back online, which will inevitably fail with a session expired error. 

I think that we should clean this up by:

1. Call `subscribeToNetInfo in `Expensify.js`. Now we subscribe both our public and auth screens to NetInfo.
2. Remove `subscribeToNetInfo` from `listenForReconnect` [here](https://github.com/Expensify/App/blob/e1788975be62592cec9b85571be6500ad78f02fc/src/libs/NetworkConnection.js#L85)
3. Move this call to [onReconnect](https://github.com/Expensify/App/blob/e1788975be62592cec9b85571be6500ad78f02fc/src/libs/actions/App.js#L256-L259) to AuthScreens' `componentDidMount` method

Thoughts?

Reply:
Thanks! That's a great insight. I can give that a try and see how it works.